tools/ctl/ctl.c

```diff
@@ -15,6 +16,29 @@
 #include "ipc/stream.h"
 #include "ipc/control.h"
 
+#define BUFFER_TAG_OFFSET	0
+#define BUFFER_SIZE_OFFSET	1
+#define BUFFER_ABI_OFFSET	2
+
+struct ctl_data {
+	/* the input file name */
+	char *input_file;
+	/* point to the cached data for input/output */
+	unsigned int *buffer;
+	/* flag for input/output format, binary or CSV */
+	bool binary;
+	/* flag for input/output format, with or without abi header */
+	bool no_abi;
+};
+
+/* global control data */
+struct ctl_data ctl_data = {
+	.input_file = NULL,
+	.buffer = NULL,
+	.binary = false,
+	.no_abi = false,
+};
```

Comment:
This should be a local variable in main() and not global.

Reply:
addressed.